test/integration/full-content/full-content.spec.js

```diff
@@ -17,38 +15,19 @@ import {
 } from '@wordpress/blocks';
 import { parse as grammarParse } from '@wordpress/block-serialization-default-parser';
 import { registerCoreBlocks } from '@wordpress/block-library';
-
-const fixturesDir = path.join( __dirname, 'fixtures' );
-
-// We expect 4 different types of files for each fixture:
-//  - fixture.html            : original content
-//  - fixture.parsed.json     : parser output
-//  - fixture.json            : blocks structure
-//  - fixture.serialized.html : re-serialized content
-// Get the "base" name for each fixture first.
-const fileBasenames = uniq(
-	fs.readdirSync( fixturesDir )
-		.filter( ( f ) => /(\.html|\.json)$/.test( f ) )
-		.map( ( f ) => f.replace( /\..+$/, '' ) )
-);
-
-function readFixtureFile( filename ) {
-	try {
-		return fs.readFileSync(
-			path.join( fixturesDir, filename ),
-			'utf8'
-		);
-	} catch ( err ) {
-		return null;
-	}
-}
-
-function writeFixtureFile( filename, content ) {
-	fs.writeFileSync(
-		path.join( fixturesDir, filename ),
-		content
-	);
-}
+import { //eslint-disable-line no-restricted-syntax
```

Comment:
I don't think it was a legitimate use-case to disable the rule here. If it's part of the package's public interface, it should be exposed from the root. We shouldn't ever need to path-traverse a module to import.

Reply:
Agreed in general, however as explained I did it on purpose to avoid exposing those methods at the moment. We will have to maintain them forever so let's test them with our own e2e tests first.

I also left this note:

> I think that we should simplify our setup by replacing all the tests in `test/integration/full-content/full-content.spec.js` with a corresponding set of e2e tests which represent a real-life usage rather than something that produces so much hassle with running fixture cleanup and generation.

If we get rid of those integration tests we will also won't need to have those methods exposed as part of the package's public interface. There is another issue which you raised in #13583 and I think noticed in one of the similar PRs which adds a different new block. It seems like those tests are sometimes broken and don't generate the correct output which makes them only confusing. It's very hard to validate them if you don't know exactly how they should behave.